probe-rs/src/probe/stlink/mod.rs

```diff
@@ -786,27 +779,25 @@ impl<D: StLinkUsb> STLink<D> {
     fn read_mem_32bit(
         &mut self,
         address: u32,
-        length: u16,
+        data: &mut [u8],
         apsel: u8,
-    ) -> Result<Vec<u32>, DebugProbeError> {
+    ) -> Result<(), DebugProbeError> {
         log::debug!(
             "Read mem 32 bit, address={:08x}, length={}",
             address,
-            length
+            data.len()
         );
         // Maximum supported read length is 2^16 bytes.
         assert!(
-            length < (u16::MAX / 4),
-            "Maximum read length for STLink is 16'384 words"
+            data.len() <= 6144,
```

Comment:
```suggestion
            data.len() <= STLINK_MAX_READ_LEN,
```

Reply:
Also, are you sure this shouldn't be divided by 4 like before and in code below?